### PR TITLE
feat: image search

### PR DESCRIPTION
**Pull request by aleksasiriski**
Closes https://github.com/hearchco/frontend/issues/119

Demo: https://feat-image-search.hearchco-frontend.pages.dev

Image search is now a category inside `/search` route that has a different rendering of UI

Images:
- [x] Searching images and proper rendering of UI
  - Needs https://github.com/hearchco/hearchco/pull/226)
- [x] Buttons for picking the category alongside bangs!
- [x] Nice loading animation for images
  - [x] Random widths
- [x] Preview when clicking on images
- [x] Good mobile UI/UX
  - [x] Dark theme toggle
  - [x] Images & Image preview
- [x] Proxy for loading images to increase user privacy
  - Needs https://github.com/hearchco/hearchco/pull/260

General & Images:
- [x] Minimum delay for loading animation (500ms)
- New requests for additional pages / results
  - Needs https://github.com/hearchco/hearchco/pull/238
  - [x] General results should have pages numbers at the bottom
  - [x] Image results should have "Show more" button to load new images and attach them to the previous ones
- [x] Dark theme toggle
- [x] Rename env from API_URL to API_URI, and added trailing `/` detection
- [x] Properly present backend errors on frontend (w/o crashing)
- [x] Read start/pages param from URL and put into variable

Refactor:
- [x] Better folder structure
- [x] Consistent order of imports with comments for clarity
- [x] Svelte formatter applied everywhere
- [x] Moved some code into separate files (components, functions, types)
- [x] Moved primary and secondary colors to tailwindcss config

### CODE REVIEW

**Review by matijakljajic — DISMISSED** [review]
LGTM

File: src/lib/components/DarkModeToggle.svelte [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		if (darkMode) {
+			document.documentElement.classList.add('dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+		}
+	}
+</script>
+
+<button on:click={toggleDarkMode}>
+	<div class="absolute top-4 right-4">
```
Comment (aleksasiriski) [inline src/lib/components/DarkModeToggle.svelte:40]: move class into button

File: src/lib/components/DarkModeToggle.svelte [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		{#if darkMode}
+			<svg
+				width="40"
+				height="40"
+				viewBox="0 0 80 80"
```
Comment (aleksasiriski) [inline src/lib/components/DarkModeToggle.svelte:45]: 0 0 40 40?

File: src/lib/components/DarkModeToggle.svelte [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+					d="M36.4733 29.1459C38.7655 28.4011 41.2345 28.4011 43.5267 29.1459C45.8189 29.8907 47.8164 31.342 49.2331 33.2918C50.6497 35.2416 51.4127 37.5899 51.4127 40C51.4127 42.4101 50.6497 44.7584 49.2331 46.7082C47.8164 48.658 45.8189 50.1093 43.5267 50.8541C41.2345 51.5989 38.7655 51.5989 36.4733 50.8541C34.1811 50.1093 32.1836 48.658 30.7669 46.7082C29.3503 44.7584 28.5873 42.4101 28.5873 40C28.5873 37.5899 29.3503 35.2416 30.7669 33.2918C32.1836 31.342 34.1811 29.8907 36.4733 29.1459Z"
+					fill="#ffb280"
+					stroke="#ffb280"
+					stroke-width="4"
+					stroke-linecap="square"
+					stroke-linejoin="round"
+				/>
+			</svg>
+		{:else}
+			<svg
```
Comment (aleksasiriski) [inline src/lib/components/DarkModeToggle.svelte:69]: move into svg files

File: src/routes/healthz/api/+server.ts [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
 export const GET: RequestHandler = async ({ fetch }) => {
-	const apiUrl = `${env.API_URL}/healthz`;
+	const apiUrl = createApiUrl('healthz');
 	const response = await fetch(apiUrl);
+
 	if (response.status >= 400 && response.status <= 599) {
-		// todo: convert number to NumericRange
-		const status: NumericRange<400, 599> = 500;
-		throw error(status, 'ERROR');
+		throw error(response.status, response.statusText);
```
Comment (aleksasiriski) [inline src/routes/healthz/api/+server.ts:11]: Does this crash frontend if backend doesn't work?

File: src/lib/functions/fetchResults.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { env } from '$env/dynamic/private';
+import { createApiUrl } from './createApiUrl';
```
Comment (aleksasiriski) [inline src/lib/functions/fetchResults.ts:1]: Switch to `$lib/functions/createApiUrl`

**Review by matijakljajic — APPROVED** [review]

**Pull request merged**